tree/forest/v7/inc/ROOT/RColumnModel.hxx

```diff
@@ -0,0 +1,60 @@
+/// \file ROOT/RColumnModel.hxx
+/// \ingroup Forest ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2018-10-09
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2015, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RColumnModel
+#define ROOT7_RColumnModel
+
+#include <ROOT/RStringView.hxx>
+
+namespace ROOT {
+namespace Experimental {
+
+// clang-format off
+/**
+\class ROOT::Experimental::EColumnType
+\ingroup Forest
+\brief The possible trivial, native content types of a column
+*/
+// clang-format on
+enum class EColumnType {
+   kUnknown,
+   kIndex, // type for root columns of nested collections
+   kByte,
+   kReal64,
+   kReal32,
+   kReal16,
+   kReal8,
+   kInt64,
+   kInt32,
+   kInt16,
+   //...
+};
```

Comment:
How will this handle general C++ classes? With `kUnknown`?

Reply:
More complex types, such as classes, get translated into columns of these simple types by the RBranch class.  I'll add a comment to the enum.

Reply:
What about object that can not (or the user does not want to) be split?